src/hotkeys_provider.rs

```diff
@@ -125,72 +116,69 @@ pub fn HotkeysProvider(
         });
     }
 
-    #[cfg(any(feature = "hydrate", feature = "csr"))]
-    div()
-        .on_mount(move |_| {
-            let blur_listener = Closure::wrap(Box::new(move || {
-                if cfg!(feature = "debug") {
-                    logging::log!("Window lost focus");
-                }
-                pressed_keys.set(HashMap::new());
-            }) as Box<dyn Fn()>);
-
-            let keydown_listener = Closure::wrap(Box::new(move |event: KeyboardEvent| {
-                pressed_keys.update(|keys| {
-                    keys.insert(event.key().to_lowercase(), event);
-                });
-            }) as Box<dyn Fn(_)>);
-            let keyup_listener = Closure::wrap(Box::new(move |event: KeyboardEvent| {
-                pressed_keys.update(|keys| {
-                    keys.remove(&event.key().to_lowercase());
-                });
-            }) as Box<dyn Fn(_)>);
+    node_ref.on_load(move |_| {
+        let blur_listener = Closure::wrap(Box::new(move || {
+            if cfg!(feature = "debug") {
+                logging::log!("Window lost focus");
+            }
+            pressed_keys.set(HashMap::new());
```

Comment:
To avoid rerenders you should probably use `set_untracked`, see https://docs.rs/leptos/latest/leptos/trait.SignalSetUntracked.html#tymethod.set_untracked